subsys/jwt/jwt.c

```diff
@@ -127,15 +127,20 @@ static int base64_append_bytes(const char *bytes, size_t len,
 	return 0;
 }
 
-struct jwt_header {
-	char *typ;
-	char *alg;
-};
-
-static struct json_obj_descr jwt_header_desc[] = {
-	JSON_OBJ_DESCR_PRIM(struct jwt_header, alg, JSON_TOK_STRING),
-	JSON_OBJ_DESCR_PRIM(struct jwt_header, typ, JSON_TOK_STRING),
-};
+/*
+ * Pre-computed JWT header
+ * Use https://www.base64encode.org/ for update
+ */
+static const char jwt_header[] =
+#ifdef CONFIG_JWT_SIGN_RSA
+	/* {"alg":"RS256","typ":"JWT"} */
+	"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9"
+#endif
+#ifdef CONFIG_JWT_SIGN_ECDSA
+	/* {"alg":"ES256","typ":"JWT"} */
+	"eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9"
+#endif
+;
```

Comment:
you could keep it in `jwt_add_header`, but no complain about put it here.

Reply:
@ceolin moved to function as you suggested